tensorboard/plugins/projector/vz_projector/data.ts

```diff
@@ -370,6 +365,91 @@ export class DataSet {
     });
   }
 
+  /** Runs UMAP on the data. */
+  async projectUmap(nComponents: number, nNeighbors: number, stepCallback: (iter: number) => void) {
+    this.hasUmapRun = true;
+    this.umap = new UMAP({nComponents, nNeighbors});
+
+    let currentEpoch = 0;
+    const epochStepSize = 10;
+    const sampledIndices = this.shuffledDataIndices.slice(0, UMAP_SAMPLE_SIZE);
+  
+    const sampledData = sampledIndices.map(i => this.points[i]);
+    // TODO: Switch to a Float32-based UMAP internal
+    const X = sampledData.map(x => Array.from(x.vector));
+
+    this.nearest = await this.computeKnn(sampledData, nNeighbors);
+    
+    const nEpochs = await util.runAsyncTask('Initializing UMAP...', () => {
+      const knnIndices = this.nearest.map(row => row.map(entry => entry.index));
+      const knnDistances = this.nearest.map(row => row.map(entry => entry.dist));
```

Comment:
nit: Please wrap at 80 like the rest of this file; Prettier suggests:

```typescript
      const knnDistances = this.nearest.map((row) =>
        row.map((entry) => entry.dist),
      );
```

Reply:
oh, answered my question re: prettier